### ISSUE TITLE
always get the error : ninja : error: loading build.ninja : No such file or directory.

### ISSUE THREAD

**Issue by pducharme** [zunichky/chromium_aaos#1]
Hi, trying to build this and got the error : 

error : ninja : error: loading build.ninja : No such file or directory.

I did all the steps from Chromium build instruction, then ran the Pull_Latest, copy the args for the release.  When trying to run Build_release.sh, I get that error.

Please help, i'm at the last step to be able to build it ! :)

**Comment by zunichky** [comment]
Do you happen to be attempting on macOs?

**Comment by pducharme** [comment]
No, I did this on Ubuntu 24.04.1 LTS. I built a VM with that OS and ran everything in this Ubuntu.

**Comment by zunichky** [comment]
I would try to run through these instructions to see if you get the error
https://chromium.googlesource.com/chromium/src/+/main/docs/android_build_instructions.md
I'm wondering if maybe something failed here: build/install-build-deps.sh --android or maybe do sudo apt-get install ninja-build

**Comment by aissagholian** [comment]
i am getting the same error now, did you find a solution?

**Comment by pducharme** [comment]
> i am getting the same error now, did you find a solution?

I used Ubuntu 22.04 LTS Instead and it was working.

**Comment by aissagholian** [comment]
really? ok let me try that and see if theres a difference

also, i am not getting the same file structure as described in the repository. is that fine?

i have /chromium and /chromium_aaos in my /home dir.

**Comment by pducharme** [comment]
> really? ok let me try that and see if theres a difference
> 
> also, i am not getting the same file structure as described in the repository. is that fine?
> 
> i have /chromium and /chromium_aaos in my /home dir.

I cant remember, I compiled a while ago and didn’t touch it since and my VM was deleted.

**Comment by aissagholian** [comment]
okay then i must have done something wrong

i am just going to start over once again using 22.04,

i'll keep you guys posted

**Comment by aissagholian** [comment]
okay i redid everything and i am still getting the "ninja: Entering directory `out/Release_arm64'
ninja: error: loading 'build.ninja': No such file or directory" error

i am on 22.04 ubuntu

**Comment by zunichky** [comment]
I can look into this more tomorrow. I was able to build Firefox and the process is much simpler than chrome if you would like me to post those instructions

**Comment by aissagholian** [comment]
firefox would be good as well. i appreciate the help. 

and yes, i would appreciate it if you can post those instructions as well. i am still awaiting my verification on google play console so i have to wait anyway.

but if patrick was able to get past this problem then i should be able to as well so hopefully he can respond with the solution too.

**Comment by aissagholian** [comment]
i used command: "rm -rf out " before running build_release.sh script again and it fixed the previous problem....

....but now i am getting this error: 
"ninja: Entering directory `out/Release_arm64'
ninja: fatal: chdir to 'out/Release_arm64' - No such file or directory"

**Comment by zunichky** [comment]
Try to run 
gn gen out/Release_arm64
Before running the script

**Comment by aissagholian** [comment]
if i do that i get this error ( fyi i am in the /chromium dir): 

ERROR Can't find source root.
I could not find a ".gn" file in the current directory or any parent,
and the --root command-line argument was not specified.

**Comment by aissagholian** [comment]
Hey @zunichky do you still intend on posting the Firefox instructions by chance?

**Comment by zunichky** [comment]
> Hey @zunichky do you still intend on posting the Firefox instructions by chance?

Here's some very quick instructions I wrote up. Sorry, they are pretty vague and I am hoping to write some in more detail at a later date. The build process in firefox is simpler/quicker so hopefully I'll be able to answer any questions 
https://github.com/zunichky/firefox_aaos

**Comment by aissagholian** [comment]
@zunichky thanks for this. i will be giving it a shot :)

**Comment by aissagholian** [comment]
so i think i did everything right but i dont know how to do the android studio things that need to be done
